src/LinearFractional.jl

```diff
@@ -1,130 +1,340 @@
-        con = LinearConstraint(LinearFractionalAffExpr([var], [1], [0]), lb, ub)
-        addconstraint(model, con)
+    inner_vref = JuMP.add_variable(m.model, v, name)
+    vref = LinearFractionalVariableRef(m, inner_vref)
+
+    if has_lower_bound(inner_vref)
+        lb = lower_bound(inner_vref)
+        delete_lower_bound(inner_vref)
+        set_lower_bound(vref, lb)
+    end
+
+    if has_upper_bound(inner_vref)
+        ub = upper_bound(inner_vref)
+        delete_upper_bound(inner_vref)
+        set_upper_bound(vref, ub)
     end
-    return LinearFractionalVariable(model, var)
+
+    return vref
+
+end
+function JuMP.delete(model::LinearFractionalModel, vref::LinearFractionalVariableRef)
+    JuMP.delete(model.model, vref.vref)
+end
+function JuMP.is_valid(model::LinearFractionalModel, vref::LinearFractionalVariableRef)
+    return (model === vref.model &&
+            is_valid(model.model, vref.vref))
+end
+
+@forward LinearFractionalModel.model JuMP.num_variables
+
+# Internal functions
+@forward LinearFractionalVariableRef.vref lower_bound,
+   has_lower_bound,
+   set_lower_bound,
+   delete_lower_bound,
+   has_upper_bound,
+   set_upper_bound,
+   delete_upper_bound,
+   upper_bound,
+   is_fixed,
+   fixed_value,
+   fix,
+   unfix,
+   start_value,
+   set_start_value,
+   set_binary,
+   unset_binary,
+   is_binary,
+   is_integer,
+   set_integer,
+   unset_integer,
+   owner_model,
+   _assert_isfinite
+
+# Defined just to be forwarded above
+variable_info(vref::VariableRef) = vref.model.variables[vref.idx].info
+function update_variable_info(vref::VariableRef, info::JuMP.VariableInfo)
+   vref.model.variables[vref.idx] = JuMP.ScalarVariable(info)
+end
+
+function set_lower_bound(vref::LinearFractionalVariableRef, lower::Number)
+    @constraint(vref.model, vref >= lower)
+end
+
+function set_upper_bound(vref::LinearFractionalVariableRef, upper::Number)
+    @constraint(vref.model, vref <= upper)
+end
+
+
+# TODO: remove after https://github.com/JuliaOpt/JuMP.jl/pull/1935 released
+function set_standard_form_rhs(
+    constraint::ConstraintRef{JuMP.Model, JuMP._MOICON{F, S}}, value) where {
+        T,
+        S <: Union{MOI.LessThan{T}, MOI.GreaterThan{T}, MOI.EqualTo{T}},
+        F <: Union{MOI.ScalarAffineFunction{T}, MOI.ScalarQuadraticFunction{T}}}
+    MOI.set(JuMP.owner_model(constraint), MOI.ConstraintSet(), constraint,
+            S(convert(T, value)))
+    return
+end
+
+
+# TODO: remove after https://github.com/JuliaOpt/JuMP.jl/pull/1935 released
+function standard_form_rhs(
+    constraint::ConstraintRef{JuMP.Model, JuMP._MOICON{F, S}}) where {
+        T,
+        S <: Union{MOI.LessThan{T}, MOI.GreaterThan{T}, MOI.EqualTo{T}},
+        F <: Union{MOI.ScalarAffineFunction{T}, MOI.ScalarQuadraticFunction{T}}}
+    con = JuMP.constraint_object(constraint)
+    return MOIU.getconstant(con.set)
 end
 
+function transform_constraint(model::LinearFractionalModel, constraint_ref::ConstraintRef)
```

Comment:
I think it's implied by the TODO above the function `set_standard_form_rhs` already since if we remove that function, the `transform_constraint` won't run at all without change.  (The change being to import the new functions from JuMP)